### ISSUE TITLE
Performance improvements for blog app

### ISSUE THREAD

**Issue by ebidel** [PolymerElements/carbon-elements#3]
TL;DR: 
@robdodson and I have been playing around with making our [blog template](http://polymerlabs.github.io/app-layout/templates/publishing/#/art/list) faster to meet RAIL's 1s first paint and to see if the blog app could be a potentially candidate for the app-shell concept.

 I prototyped http://ebidel.github.io/app-layout/templates/publishing/#/art/list with some performance improvements.

*Results - Nexus 5 - Chrome - 3GFast)*:
  - Original -  load: 2.5s, first paint 3.325s [results](http://www.webpagetest.org/result/151028_32_5CP/1/details/)
  - Improved - load: 2.2s, first paint 2.860s [results](http://www.webpagetest.org/result/151028_5G_6T3/)  **~500ms improved first paint**

===

What I did:
- `<script src="webcomponents-lite.js" async>` - don't block render!
- Moved imports into an elements.html import async: `< link rel="elements.html" async>` - don't block render!
- removed `<body unresolved>` - dont block render!
- moved markup of `<blog-app>` into a `<template is="dom-bind">` in the main page. This gives something to paint faster and doesn't delay on slow loading element upgrades.
- used vulcanize w/ `crisper --script-in-head` - defer loads the elements.v.js bundle at the top of the elements.v.html bundle. For large vulcanized bundles, the parsers has to go through a ton of svg/html/css before it starts loading elements.v.js (and therefore, elements get rendered later). Ideally we could even move the script load into the main page, but there are timing issues with this. This change alone can [speed up the vulcanized script download by 80%](https://github.com/PolymerLabs/crisper/issues/22).
- It's actually better not to vulcanize index.html itself - this increases the weight of the main page quite a big, inling all resources. On slow connections this can be veery noticeable.
- removed `iron-ajax` in favor of  `fetch()`/xhr'ing on the main page - this kicks off the data fetch for articles.json sooner. The data is ready to go by the time the app is upgraded! Otherwise, iron-ajax is waiting to upgrade before it makes the fetch. This pushes out the network timeline.

<img width="961" alt="screen shot 2015-10-30 at 10 45 11 am" src="https://cloud.githubusercontent.com/assets/238208/10853447/52ba6c78-7ef3-11e5-94ae-6de87cd12c98.png">

Still....
**The largest source of rendering delay is from the `<blog-app>` or `<template is="dom-bind">` pattern.  Hopefully https://github.com/Polymer/polymer/issues/2654 can help address alternatives for people that want really fast 1st renders.**

**Comment by tjsavage** [comment]
Closing here for now
